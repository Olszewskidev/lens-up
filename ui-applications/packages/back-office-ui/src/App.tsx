import { RouterProvider } from 'react-router-dom'
import './App.css'
import AppRouter from './pages/AppRouter'

function App() {
  return (
    <RouterProvider router={AppRouter} />
  )
}

export default App
