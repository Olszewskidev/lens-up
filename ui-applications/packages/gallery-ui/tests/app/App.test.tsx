import { render, screen } from '@testing-library/react'
import { vi, expect, test } from 'vitest'
import App from '../../src/App';
import { createAsyncThunk } from '@reduxjs/toolkit';
import { create } from './store';

/**
* @vitest-environment jsdom
*/
test('Middleware must log api failure', async () => {
    const { next, invoke } = create();
    const consoleMock = vi.spyOn(console, 'error').mockImplementation(() => undefined);
    const action = createAsyncThunk("test/accepted", async (state: string, { rejectWithValue }) => {
        if (state != "accepted")
          return rejectWithValue("Console must read api failure");
      });
    invoke(action);
    expect(consoleMock).toHaveBeenCalledOnce();
    expect(consoleMock).toHaveBeenLastCalledWith('API Failed!');
  });

/**
* @vitest-environment jsdom
*/
test('App must render according to snapshot', () => {
    const {getByText, asFragment} = render(
        <App />,
    );

    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });