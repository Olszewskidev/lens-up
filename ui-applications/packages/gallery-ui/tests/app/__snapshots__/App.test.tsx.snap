// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App must render according to snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-black"
  >
    <div
      class="container mx-auto flex flex-wrap flex-col md:flex-row justify-center h-screen items-center"
    >
      <div>
        <img
          alt="LensUp logo."
          height="147"
          src="/@fs/C:/Users/safronov/source/lens-up/ui-applications/packages/shared-components/src/images/lens-up-logo.png"
          width="379"
        />
        <form
          class="px-7 py-4 bg-black rounded-lg leading-none border-double border-4 border-white shadow-lg max-w-sm mx-auto"
        >
          <div
            class="mb-5"
          >
            <div>
              <label
                class="block mb-2 text-sm font-medium text-gray-50 dark:text-white"
                for=":r0:"
              >
                Join to your gallery:
              </label>
              <input
                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                id=":r0:"
                placeholder="Enter code"
                required=""
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="flex items-center justify-center"
          >
            <div
              class="text-center relative group"
            >
              <button
                class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-full text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 inline-flex items-center justify-center"
                type="submit"
              >
                Join
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
